src/Core/PluginCollection.php

```diff
@@ -78,6 +79,49 @@ public function __construct(array $plugins = [])
         $this->loadConfig();
     }
 
+    /**
+     * Add plugins from config array.
+     *
+     * @param array $config Configuration array. For e.g.:
+     *   ```
+     *   [
+     *       'Company/TestPluginThree',
+     *       'TestPlugin' => ['onlyDebug' => true, 'onlyCli' => true],
+     *       'Nope' => ['optional' => true],
+     *       'Named' => ['routes' => false, 'bootstrap' => false],
+     *   ]
+     *   ```
+     * @return void
+     */
+    public function addFromConfig(array $config): void
+    {
+        $debug = Configure::read('debug');
+        $cli = PHP_SAPI === 'cli';
+
+        foreach (Hash::normalize($config) as $name => $options) {
+            $options = (array)$options;
+            $onlyDebug = $options['onlyDebug'] ?? false;
```

Comment:
This could just be `debug` and `cli`.

Reply:
Adding the `only` suffix makes them more readable and intuitive IMO. The plugin load command will correspondingly have `--only-debug` option.

Reply:
Sounds good. At first `onlyDebug` sounded odd compared to `debugOnly`, but it reads correctly in a sentence. Either way.

Reply:
I did flip flop between having "only" as suffix or prefix but it felt better as prefix. Open to changing it to suffix if that's preferred by everyone.

Reply:
:+1: for the only prefix. It makes it clearer what the key does and when it is relevant.